daemons/fenced/fenced_cib.c

```diff
@@ -365,124 +365,94 @@ fenced_query_cib(void)
-
-        xml_patch_versions(patchset, add, del);
-
-        for (change = pcmk__xe_first_child(patchset, NULL, NULL, NULL);
-             change != NULL; change = pcmk__xe_next(change)) {
-
-            const char *op = crm_element_value(change, PCMK_XA_OPERATION);
-            const char *xpath = crm_element_value(change, PCMK_XA_PATH);
+    int format = 1;
 
-            if(op == NULL) {
-                continue;
+    int add[] = { 0, 0, 0 };
+    int del[] = { 0, 0, 0 };
 
-            } else if(strstr(xpath, "/" PCMK_XE_FENCING_LEVEL) != NULL) {
-                /* Change to a specific entry */
+    CRM_CHECK(patchset != NULL, return);
 
-                crm_trace("Handling %s operation %d.%d.%d for %s", op, add[0], add[1], add[2], xpath);
-                if (strcmp(op, PCMK_VALUE_MOVE) == 0) {
-                    continue;
+    crm_element_value_int(patchset, PCMK_XA_FORMAT, &format);
+    if (format != 2) {
+        crm_warn("Unknown patch format: %d", format);
+        return;
+    }
 
-                } else if (strcmp(op, PCMK_VALUE_CREATE) == 0) {
-                    add_topology_level(change->children);
+    xml_patch_versions(patchset, add, del);
 
-                } else if (strcmp(op, PCMK_VALUE_MODIFY) == 0) {
-                    xmlNode *match = pcmk__xe_first_child(change,
-                                                          PCMK_XE_CHANGE_RESULT,
-                                                          NULL, NULL);
+    for (xmlNode *change = pcmk__xe_first_child(patchset, NULL, NULL, NULL);
+         change != NULL; change = pcmk__xe_next(change)) {
 
-                    if(match) {
-                        remove_topology_level(match->children);
-                        add_topology_level(match->children);
-                    }
+        const char *op = crm_element_value(change, PCMK_XA_OPERATION);
+        const char *xpath = crm_element_value(change, PCMK_XA_PATH);
 
-                } else if (strcmp(op, PCMK_VALUE_DELETE) == 0) {
-                    /* Nuclear option, all we have is the path and an id... not enough to remove a specific entry */
-                    crm_info("Re-initializing fencing topology after %s operation %d.%d.%d for %s",
-                             op, add[0], add[1], add[2], xpath);
-                    fencing_topology_init();
-                    return;
-                }
+        if (op == NULL) {
+            continue;
+        }
 
-            } else if (strstr(xpath, "/" PCMK_XE_FENCING_TOPOLOGY) != NULL) {
-                /* Change to the topology in general */
-                crm_info("Re-initializing fencing topology after top-level %s operation  %d.%d.%d for %s",
+        if (strstr(xpath, "/" PCMK_XE_FENCING_LEVEL) != NULL) {
+            // Change to a specific entry
+            crm_trace("Handling %s operation %d.%d.%d for %s",
+                      op, add[0], add[1], add[2], xpath);
+
+            if (strcmp(op, PCMK_VALUE_DELETE) == 0) {
```

Comment:
what happened to PCMK_VALUE_MOVE?

Reply:
We can keep it for completeness if we want, but the behavior is the same either way. There's no catch-all "else" case, only "else if"s.

Reply:
Er, there's a `continue` at the bottom.

Reply:
I'll double-check this and get back to you.

Reply:
yeah that seems to cover it